test/blackbox-tests/test-cases/pkg/depopts/depopts-with-constraints.t

```diff
@@ -0,0 +1,38 @@
+We test how the solver chooses a solution for depopts with constraints.
+
+  $ . ../helpers.sh
+  $ mkrepo
+
+We create a package "foo" that is the dependency of two packages "optional" and
+"bar".
+
+  $ mkpkg foo 1
+
+The package "optional" has a constraint on a particular version of "foo". 
+
+  $ mkpkg optional <<'EOF'
+  > depends: [ "foo" {= "1"} ]
+  > EOF
+
+The package "bar" depends on "foo" and depends optionally on "optional".
+
+  $ mkpkg bar <<'EOF'
+  > depends: [ "foo" ]
+  > depopts: [ "optional" ]
+  > EOF
+
+Here we expect the solver to choose all 3 packages as a build plan.
```

Comment:
We have no such expectation. It's within our right to ignore the optional dependency. It's best to describe what we're doing here rather than speculating how it should work.

Reply:
I've updated the comment.